src/pipeline/node/NeuralNetwork.cpp

```diff
@@ -38,6 +38,21 @@ void NeuralNetwork::setBlob(OpenVINO::Blob blob) {
     properties.blobSize = static_cast<uint32_t>(asset->data.size());
 }
 
+void NeuralNetwork::setXmlModelPath(const dai::Path& xmlModelPath, const dai::Path& binModelPath) {
+    auto xmlAsset = assetManager.set("__xmlModel", xmlModelPath);
+    dai::Path localBinModelPath;
+    if(!binModelPath.empty()) {  // Path for the bin file IS set
+        localBinModelPath = binModelPath;
+    } else {  // Path for the bin file IS NOT set
+        auto lastDotPos = xmlModelPath.string().find_last_of('.');
+        localBinModelPath = xmlModelPath.string().substr(0, lastDotPos) + ".bin";
+    }
+    auto binAsset = assetManager.set("__binModel", localBinModelPath);
+    properties.xmlUri = xmlAsset->getRelativeUri();
+    properties.binUri = binAsset->getRelativeUri();
+    properties.modelSource = NeuralNetworkProperties::ModelSource::XML_API;
```

Comment:
```suggestion
    properties.modelSource = Properties::ModelSource::XML;
```

:+1: 
But setBlobPath is missing the same call but setting back to `Properties::ModelSource::BLOB;`